pyrenew_hew/pyrenew_hew_data.py

```diff
@@ -118,12 +132,179 @@ def n_days_post_init(self):
             self.last_data_date_overall - self.first_data_date_overall
         ).days
 
+    @property
+    def site_subpop_spine(self):
+        ww_data_present = self.data_observed_disease_wastewater is not None
+        if ww_data_present:
+            # Check if auxiliary subpopulation needs to be added
+            add_auxiliary_subpop = (
+                self.population_size
+                > self.data_observed_disease_wastewater.select(
+                    pl.col("site_pop", "site", "lab", "lab_site_index")
+                )
+                .unique()
+                .sum()
+                .to_numpy()
+                .flatten()[0]
+            )
+            site_indices = (
+                self.data_observed_disease_wastewater.select(
+                    ["site_index", "site", "site_pop"]
+                )
+                .unique()
+                .sort("site_index")
+            )
+            if add_auxiliary_subpop:
+                aux_subpop = pl.DataFrame(
+                    {
+                        "site_index": [None],
+                        "site": [None],
+                        "site_pop": [
+                            self.population_size
+                            - site_indices.select(pl.col("site_pop"))
+                            .sum()
+                            .to_numpy()
+                            .flatten()[0]
+                        ],
+                    }
+                )
+            else:
+                aux_subpop = pl.DataFrame()
+            site_subpop_spine = (
+                pl.concat([aux_subpop, site_indices], how="vertical_relaxed")
+                .with_columns(
+                    [
+                        pl.col("site_index").cum_count().alias("subpop_index"),
+                        pl.when(pl.col("site").is_not_null())
+                        .then(
+                            pl.col("site").map_elements(
+                                lambda x: f"Site: {x}", return_dtype=str
```

Comment:
Can you use something more optimized from [`polars.Series.string`](https://docs.pola.rs/api/python/stable/reference/series/string.html)?